src/PoolManager.sol

```diff
@@ -149,11 +149,11 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, Claims {
     }
 
     /// @inheritdoc IPoolManager
-    function lock(bytes calldata data) external override returns (bytes memory result) {
-        Lockers.push(msg.sender);
+    function lock(address lockTarget, bytes calldata data) external payable override returns (bytes memory result) {
```

Comment:
did you consider adding a `if (msg.value > 0) settle()` to the lock function?
definitely not needed but it would probably bring down gas further for our ETH input cases (which is a huge % of our volume)

Reply:
it just feels weird that the ETH is sent in this function but is actually used in a different functions